### PR TITLE
[GEOS-7676] Makes the importer share the request context when starting a job asynchronously

### PR DESCRIPTION
**Pull request by nmco**
Associated issue: https://osgeo-org.atlassian.net/browse/GEOS-7676

This pull request also provides a test case for this.

### CODE REVIEW

File: src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.geoserver.security.AbstractResourceAccessManager;
+import org.geoserver.security.DataAccessLimits;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The propose of this class is to test that requests spring context are correctly shared
```
Comment (aaime) [inline src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java]: purpose

**Comment by aaime** [comment]
Tiny typo fix needed, then good to go, merge at will. Mind you used java 8 syntax so if you need to backport the fix to 2.8.x, it will have to be amended

File: src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java (hunk 1 above)
Comment (nmco) [inline src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java]: Well it was the only way I found to test that the spring context was share between several threads.

Comment (nmco) [inline src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java]: haha ok I got it sorry fixing the typo ...

**Comment by nmco** [comment]
Type fixed, thanks for the review.

**Pull request merged**